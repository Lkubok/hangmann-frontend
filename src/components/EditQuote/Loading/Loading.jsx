import React, { Component } from "react";
import { css } from "@emotion/core";
import { GridLoader } from "react-spinners";
import "./Loading.scss";

const override = css`
  display: block;
  margin: 0 auto;
  border-color: red;
  color: green;
`;

export default class Loading extends Component {
  state = {
    dots: ""
  };
  componentDidMount() {
    this.interval = setInterval(() => {
      this.setState(prevState => {
        return {
          dots: prevState.dots + "."
        };
      });
    }, 250);
  }
  componentWillUnmount() {
    if (this.interval) {
      clearInterval(this.interval);
    }
  }

  render() {
    return (
      <div className="loading-holder">
        <div className="sweet-loading">
          <GridLoader
            css={override}
            sizeUnit={"px"}
            size={25}
            color={"#6dba0a"}
            loading={this.state.loading}
          />
        </div>
      </div>
    );
  }
}
