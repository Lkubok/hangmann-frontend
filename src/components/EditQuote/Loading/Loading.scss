@import "../../../scss/mixins.scss";
@import "../../../scss/variables.scss";

.loading-holder {
  @include flex-cc-col;
  @include content-base-container;
  width: 100%;

  flex-basis: 100%;
  color: $primary-color;
}
