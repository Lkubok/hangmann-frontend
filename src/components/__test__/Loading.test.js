import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Loading } from "../EditQuote/Loading/Loading";
Enzyme.configure({ adapter: new Adapter() });

describe("Loading component", () => {
  beforeEach(() => {});
  it("Should match snapshot", () => {
    const wrapper = shallow(<Loading />);
    expect(wrapper).toMatchSnapshot();
  });
});
