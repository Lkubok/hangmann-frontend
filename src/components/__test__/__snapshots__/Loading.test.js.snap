// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component Should match snapshot 1`] = `
<div
  className="loading-holder"
>
  <div
    className="sweet-loading"
  >
    <Loader
      color="#6dba0a"
      css={
        Object {
          "map": undefined,
          "name": "mso4ej",
          "next": undefined,
          "styles": "
  display: block;
  margin: 0 auto;
  border-color: red;
  color: green;
",
          "toString": [Function],
        }
      }
      loading={true}
      margin="2px"
      size={25}
      sizeUnit="px"
    />
  </div>
</div>
`;
